clients/python/src/model_registry/_client.py

```diff
@@ -87,11 +87,78 @@ def __init__(
                 server_address, port, user_token
             )
 
-    def async_runner(self, coro: Any) -> Any:
+    @classmethod
+    def from_service(
+        cls, name: str, author: str, *, ns: str | None = None, is_secure: bool = True
+    ) -> ModelRegistry:
+        """Create a client from a service name.
+
+        Args:
+            name: Service name.
+            author: Name of the author.
+
+        Keyword Args:
+            ns: Namespace. Defaults to DSC registriesNamespace, or `kubeflow` if unavailable.
+            is_secure: Whether to use a secure connection. Defaults to True.
+        """
+        from kubernetes import client, config
+
+        config.load_incluster_config()
+        if not ns:
+            kcustom = client.CustomObjectsApi()
+            try:
+                dsc_raw = kcustom.list_cluster_custom_object(
+                    group="datasciencecluster.opendatahub.io",
+                    version="v1",
+                    plural="datascienceclusters",
+                )
+            except client.ApiException as e:
+                msg = f"Failed to list datascienceclusters: {e}"
+                warn(msg, stacklevel=2)
+                # default
+                ns = "kubeflow"
+            else:
+                ns = t.cast(
+                    dict[str, t.Any],
+                    dsc_raw["items"][0],
+                )["status"]["components"]["modelregistry"]["registriesNamespace"]
+
+        kcore = client.CoreV1Api()
+        serv = t.cast(client.V1Service, kcore.read_namespaced_service(name, ns))
+        meta = t.cast(client.V1ObjectMeta, serv.metadata)
+        ext_addr = t.cast(dict[str, str], meta.annotations).get(
+            "routing.opendatahub.io/external-address-rest"
+        )
+        if ext_addr:
+            host, port = ext_addr.split(":")
+        elif not is_secure:
+            host = meta.name
+            port = next(
+                (
+                    str(port.port)
+                    for port in t.cast(
+                        list[client.V1ServicePort],
+                        t.cast(client.V1ServiceSpec, serv.spec).ports,
+                    )
+                    if port.app_protocol == "http"
+                ),
+                "8080",
+            )
+        else:
+            msg = "No external address found for secure connection"
+            raise StoreError(msg)
+
+        return cls(
+            f"https://{host}",
```

Comment:
I might be missing something, but I think in the case of `is_secure` this ought to be switched to `http://` prefix? what am I missing? 🤔

Reply:
oh that's true.